tests/main/snap-quota-groups/task.yaml

```diff
@@ -0,0 +1,80 @@
+summary: Functional test for quota-related snap commands.
+
+details: |
+  Functional test for snap quota group commands ensuring that they are 
+  effective in practice.
+
+prepare: |
+  if os.query is-trusty; then
+    exit 0
+  fi
+  snap install go-example-webserver jq remarshal
+  snap set system experimental.quota-groups=true
+  tests.cleanup defer snap unset system experimental.quota-groups
+
+execute: |
+  if os.query is-trusty; then
+    # just check that we can't do anything with quota groups on trusty, systemd
+    # there is 204, but we need at least 205 for slice units
+
+    snap quota foobar --max-memory=1MB 2>&1 | MATCH "systemd version too old: snap quotas requires systemd 205 and newer \(currently have 204\)"
+    exit 0
+  fi
+
+  echo "Create a group with a snap in it"
+  snap quota group-one --max-memory=100MB go-example-webserver
+
+  echo "The systemd slice should be active now"
+  sliceName="snap.$(systemd-escape --path group-one).slice"
+  systemctl show --property=ActiveState "$sliceName" | MATCH "ActiveState=active"
+
+  echo "The service should also still be active"
+  snap services go-example-webserver.webserver | MATCH "go-example-webserver.webserver\s+enabled\s+active"
+
+  echo "The systemd slice should have one process in it now"
+  cgroupProcsFile="/sys/fs/cgroup/memory/$sliceName/cgroup.procs"
```

Comment:
This path needs to be different for systems with cgroupv2, that'll be `/sys/fs/cgroup/$sliceName/cgroup.procs`

Reply:
ahhh yes that makes sense and explains some of the failures, good catch

Reply:
@mvo5 took care of this for me (thanks Michael)